### DISCUSSION TITLE
Equality and mixins

### DISCUSSION THREAD

**Discussion by cpurdy** [xtclang/xvm#28] [Ideas]
For discussion: Should annotations that do not define their own equals() function interfere with the equality contract? For example:

```
Int x = 3;
@Unchecked Int y = 3;
if (x == y) // compile time error: types do not match
    {
    //...
    }
```

**Comment by markandrewfalco** [comment]
It would seem the normal xtc contact covers this. I.e. equality is based on how you reference the objects. What is the equality rule for comparing two different types? I assume it is a compile time error.

**Comment by cpurdy** [comment]
Yes, it's a compile time error (the only allowed type difference is that one may be Nullable).

**Reply by markandrewfalco** [reply]
it seems reasonable to leave it as is then, if the user wants a different form of equality they can get it by changing the ref type
